crates/sui-sdk-types/src/types/transaction/unresolved.rs

```diff
@@ -93,28 +94,36 @@ pub struct UnresolvedObjectReference {
     serde(rename_all = "snake_case")
 )]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
-pub enum UnresolvedInputArgumentKind {
+pub enum InputArgumentKind {
     Pure,
     Shared,
     Receiving,
     ImmutableOrOwned,
-    Immutable,
-    Owned,
     Literal,
 }
 
+/// A potentially unresolved input object. Note that one can construct a fully resolved input
+/// argument using the provided constructors, but this struct is also useful when the input data is
+/// not complete.
+///
+/// If used in the context of transaction builder, make sure to call `tx.resolve` function to
+/// ensure that this input argument is attempted to be resolved (along with other unresolved
+/// transaction data that exists in the transaction builder).
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 #[cfg_attr(
     feature = "serde",
     derive(serde_derive::Serialize, serde_derive::Deserialize)
 )]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
-pub struct UnresolvedInputArgument {
+pub struct InputArgument {
+    /// The object kind.
     #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none",))]
-    pub kind: Option<UnresolvedInputArgumentKind>,
+    pub kind: Option<InputArgumentKind>,
```

Comment:
nit: Can we call these `Input` and `InputKind`? I don't think `Argument` is adding much here.

Reply:
To quote a VIP - the world is our oyster, so yes we can :D

Reply:
Renamed.